packages/user_repository/lib/src/user_repository.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:api_client/api_client.dart';
+import 'package:hub_domain/hub_domain.dart';
+
+/// Exception thrown when there is no authenticated user.
+class AuthenticationFailure implements Exception {}
+
+///  Exception thrown when there is an error getting user information.
+class UserInformationFailure implements Exception {
+  /// {@macro user_information_failure}
```

Comment:
I think you need to add the `template` annotations for `UserInformationFailure` so the `macro` annotation can reference the dartdoc.